plugins/modules/gpg_keypair.py

```diff
@@ -0,0 +1,390 @@
+
+def delete_keypair(
+    gpg_runner: PluginGPGRunner,
+    matching_keys: List[str],
+    check_mode: bool = False
+) -> Dict[str, Union[str, int]]:
+    if matching_keys:
+        gpg_runner.run_command([
+            f'{"dry-run" if check_mode else ""}',
+            '--batch',
+            '--yes',
+            '--delete-secret-and-public-key',
+            *matching_key
+        ], check_rc=True)
+        if params['return_fingerprints']:
+            return dict(changed=True, fingerprints=matching_keys)
+        return dict(changed=True, fingerprints=[])
+    return dict(changed=False, fingerprints=[])
+
+def generate_keypair(
+    gpg_runner: PluginGPGRunner,
+    params: Dict[str, Union[str, int, bool, List[str]]],
+    matching_keys,
+    check_mode: bool = False
+) -> Dict[str, Union[bool, str]]:
+    if matching_keys:
+        if params['return_fingerprints']:
+            return dict(changed=False, fingerprints=matching_keys)
+        return dict(change=False, fingerprints=[])
+
+    parameters = f"""<<EOF
+        Key-Type: {params['key_type']}
+        Key-Length: {params['key_type']}
+        Key-Curve: {params['key_curve']}
+        {f'''
+        Subkey-Type: {params["subkey_type"]}
+        Subkey-Length: {params["subkey_type"]}
+        Subkey-Curve: {params["subkey_curve"]}
+        ''' if params['subkey_type'] else ''}
+        Expire-Date: {params['expire_date']}
+        {f'Name-Real: {params["name"]}' if params['name'] else ''}
+        {f'Name-Comment: {params["comment"]}' if params['comment'] else ''}
+        {f'Name-Email: {params["email"]}' if params['email'] else ''}
+        {f'Passphrase: {params["passphrase"]}' if params['passphrase'] else '%no-protection'}
+        {f'Keyserver: {params["keyserver"]}' if params['keyserver'] else ''}
+        {'%transient-key' if params['transient_key'] else ''}
+        {'%dry-run' if check_mode or not params['force'] else ''}
+        %commit
+        EOF
+        """
+
+    _, stdout, _ = gpg_runner.run_command([
+        f'{"dry-run" if check_mode else ""}',
+        '--batch',
+        '--log-file',
+        '/dev/stdout',
+        '--gen-key',
+        f'{parameters}'
+    ])
+
+    if params['return_fingerprints']:
+        fingerprints = []
+        fingerprint = re.search(r"([a-zA-Z0-9]*)\.rev", stdout)
+        if fingerprint:
+            fingerprints.append(fingerprint)
+        return dict(changed=True, fingerprints=fingerprints)
+    return dict(changed=True, fingerprints=[])
+
+def run_module(params: Dict[str, Union[str, int, bool, List[str]]], check_mode: bool = False):
+    validate_params(params)
+    gpg_runner = PluginGPGRunner()
+    matching_keys = list_matching_keys(
+        params["name"],
+        params["comment"],
+        params["email"],
+        params["fingerprints"]
+    )
+    result = generate_keypair(gpg_runner, params, matching_keys, check_mode) if params['state'] == 'present' else delete_keypair(gpg_runner, matching_keys, check_mode)
```

Comment:
It's easier to read if you split this up:
```suggestion
    if params['state'] = present:
        result = generate_keypair(gpg_runner, params, matching_keys, check_mode)
    else:
        result = delete_keypair(gpg_runner, matching_keys, check_mode)
```